### DISCUSSION TITLE
Option to default to mo.plain for dataframes?

### DISCUSSION THREAD

**Discussion by foxcroftjn** [marimo-team/marimo#2033] [Q&A] [answered]
As the result of a recent change (#1697), the default rendering of dataframes now uses `mo.ui.table` by default. While I am pleased that `mo.plain` exists, I find that I am typing it a lot and this is creating some unpleasant friction.

I liked the previous ability to set default table display options globally. Before this change, I could add a line like `pl.Config.set_tbl_rows(50)` right after `import polars as pl` and this would set the default number of rows shown in tables throughout the entire notebook.

As best I can tell, there is no way to provide default arguments to the default `mo.ui.table` function call **(option A)**, or revert to using `mo.plain` by default instead of `mo.ui.table` via a configuration option (either at a global level or a notebook level) **(option B)**.

I would prefer option B, and am hoping for some feedback about the following:
1. Have I missed a configuration option that would solve this problem?
2. Is marimo aiming to be feature-light and opinionated, and would therefore consider an option for choosing a default between `mo.plain` and `mo.ui.table` to be bloat?
3. If the answer to the above two questions is no, would the maintainers be willing to add a configuration option to `~/.marimo.toml` to select a default between `mo.plain` and `mo.ui.table`?

I appreciate any feedback on these questions!

**Comment by akshayka** [comment]
Thanks for the thoughtful discussion.

We could consider adding a notebook configuration to default to mo.plain for dataframes. I'm not sure if it would be global or per notebook.

If we instead provided a configuration for the default number of rows shown in ui.table, would that be satisfactory?

**Reply by foxcroftjn** [reply]
A configuration option, either global or per notebook, would definitely address this to my satisfaction.

To elaborate on why I prefer `mo.plain`:
1. I make heavy use of marimo's ability to export notebooks as static HTML documents. The tables generated by `mo.ui.table` are not ideal in this context, as they contain a variety of non-functioning UI elements (i.e., download, search, everything in the menu that appears when you click a column name). When viewing the static HTML page, a user only learns that the search does not work after clicking the search button, typing a query, and pressing enter.
2. I work with very large dataframes, and there is sometimes a noticeable delay for `mo.ui.table` to generate the column summaries. When I am frequently re-running a cell while incrementally building a complex query, this delay is frustrating. I know this can be turned off via the `show_column_summaries` option, but like the `page_size` option, there is no way to set this argument globally and I don't want the friction of repeatedly wrapping dataframes queries in `mo.ui.table(..., show_column_summaries=False)` any more than `mo.plain(...)`.

Unfortunately, this means that a configuration for the default number of rows shown in ui.table would not really address my use case. I appreciate your willingness to hear my use case out! marimo is an invaluable tool to me, and I use it daily.

**Comment by akshayka** [comment, marked as answer]
@foxcroftjn — the next release will include a user configuration option to default to the plain dataframe viewer. Thanks for your patience!

See https://github.com/marimo-team/marimo/pull/2067
